### PR TITLE
[12.0][IMP] - Chronology check depends on invoice type if journal is set to Dedicated Credit Note Sequence

### PR DESCRIPTION
**Pull request by sbejaoui**

### CODE REVIEW

**Comment by sbejaoui** [comment]
cc/ @sbidoul, @gva-acsone

File: account_invoice_constraint_chronology/model/account_invoice.py [hunk 1]
```diff
@@ -24,14 +24,20 @@ def _prepare_previous_invoices_domain(self, invoice):
...
 
     @api.model
     def _prepare_later_invoices_domain(self, invoice):
-        return [
+        domain = [
             ('state', 'in', ['open', 'in_payment', 'paid']),
             ('date_invoice', '>', invoice.date_invoice),
             ('journal_id', '=', invoice.journal_id.id),
         ]
+        if invoice.journal_id.refund_sequence:
```
Comment (sbidoul) [inline account_invoice_constraint_chronology/model/account_invoice.py]: Maybe this ?

```suggestion
        if invoice.journal_id.refund_sequence and invoice.journal_id.sequence_id != invoice.journal_id.refund_sequence_id:
```

**Review by sbidoul — APPROVED** [review]
Since it only makes sense to enforce the chronology with respect to the sequence, it is useful to allow users to post a refund with a date anterior to the last invoice, when the refund sequence is different.

So LGTM (code review).

File: account_invoice_constraint_chronology/model/account_invoice.py (hunk 1 above)
Comment (sbejaoui) [reply account_invoice_constraint_chronology/model/account_invoice.py]: yes,

**Review by gva-acsone — APPROVED** [review]
LGTM Functional test

**Pull request merged**